zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/service/DocumentenApiVersionService.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.documentenapi.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.ritense.authorization.Action
+import com.ritense.authorization.AuthorizationContext
+import com.ritense.authorization.AuthorizationService
+import com.ritense.authorization.request.EntityAuthorizationRequest
+import com.ritense.document.domain.impl.JsonSchemaDocumentDefinition
+import com.ritense.document.service.DocumentDefinitionService
+import com.ritense.document.service.DocumentService
+import com.ritense.document.service.JsonSchemaDocumentDefinitionActionProvider
+import com.ritense.document.service.impl.JsonSchemaDocumentDefinitionService
+import com.ritense.document.service.impl.JsonSchemaDocumentService
+import com.ritense.documentenapi.DocumentenApiPlugin
+import com.ritense.documentenapi.domain.DocumentenApiVersion
+import com.ritense.plugin.domain.PluginConfiguration
+import com.ritense.plugin.service.PluginService
+import com.ritense.processdocument.service.DocumentDefinitionProcessLinkService
+import com.ritense.valtimo.camunda.repository.CamundaProcessDefinitionSpecificationHelper
+import com.ritense.valtimo.camunda.service.CamundaRepositoryService
+import com.ritense.valtimo.processlink.service.PluginProcessLinkService
+import org.springframework.core.io.Resource
+import org.springframework.core.io.ResourceLoader
+import org.springframework.core.io.support.ResourcePatternUtils
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+@Transactional
+class DocumentenApiVersionService(
+    private val resourceLoader: ResourceLoader,
+    private val objectMapper: ObjectMapper,
+    private val documentenApiVersions: MutableMap<String, DocumentenApiVersion>,
```

Comment:
Why is this map mutable? If it's because of the `init` block inside, it would be better to copy the map there to a `private val` and remove `private val` from this argument.

Reply:
Changed